### DISCUSSION TITLE
Direct access to global data within partial template (Liquid)

### DISCUSSION THREAD

**Discussion by SchraderMa** [11ty/eleventy#3186] [Q&A]
Hello everyone.

Is there a way to access global data (data file, process.env, etc.) within a partial template (Liquid) without passing the data over in render tag?

/_data/datafile.json
```
{
  direct: "Direct value",
  passed: "Passed value"
}
```

/page-file.html
```
{% render 'partial.html', passed: datafile.passed %}

This is working: {{ datafile.direct}}
```

/_includes/partial.html
```
This is working: {{ passed }}
This is not working: {{ datafile.direct }}
```

It would be very helpful to access global data directly in partials, otherwise you have to pass all the data through all render calls. 

Thx for your help!

Best regards,
Marc

**Comment by gabidombrowski** [comment]
Did you ever figure this out? I am having the same problem.

**Comment by dwkns** [comment]
How are you including `_includes/partial.html` into your templates?

given `_includes/partial.html`
```liquid
<p>passed = {{ passed }}</p>
<p>datafile.direct = {{ datafile.direct }}</p>
```

And `index.liquid`
```liquid
   <h1>Using Render</h1>
   {% render 'partial.html' %}

   <h1>Using Render with passed</h1>
   {% render 'partial.html', passed: datafile.passed %}
  
   <h1>Using Include</h1>
   {% include 'partial.html' %}

   <h1>Using Include with passed</h1>
   {% include 'partial.html' , passed: datafile.passed %}

   <h1>Direct</h1>
   <p>datafile.direct= {{ datafile.direct }}</p>
 ```
 You should get:
---
### Using Render

passed = 
datafile.direct =

### Using Render with passed

passed = Passed value
datafile.direct =

### Using Include

passed =
datafile.direct = Direct value

### Using Include with passed

passed = Passed value
datafile.direct = Direct value

### Direct

datafile.direct= Direct value

---

It may be worth noting that`{% Include %}` is [depreciated in Liquid](https://shopify.github.io/liquid/tags/template/#:~:text=include,template%20within%20the%20current%20template.&text=The%20include%20tag%20works%20similarly,variables%20within%20its%20parent%20template.). And passing data using `render` is the prefered way forward.
